File: frontEnd/src/app/core/models/cupon-response.ts
import { EstadoCupon } from './enums/estado-cupon.enum';
import { TipoDescuento } from './enums/tipo-descuento.enum';

export interface CuponResponse {
    id: number;
    codigo: string;
    tipoDescuento: TipoDescuento;
    valor: number;
    fechaInicio: string;
    fechaFin: string;
    usoMaximo: number | null;
    usosActuales: number;
    estado: EstadoCupon;
    /**
     * Cupos restantes para asignar a nuevos usuarios.
     * null = sin límite (usoMaximo es null).
     * 0   = agotado.
     */
    cuposDisponibles: number | null;
}

File: Backend/src/main/java/com/Trabajo_Final_Beltran/dto/response/CuponResponse.java
package com.Trabajo_Final_Beltran.dto.response;

import com.Trabajo_Final_Beltran.enums.EstadoCupon;
import com.Trabajo_Final_Beltran.enums.TipoDescuento;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.math.BigDecimal;
import java.time.LocalDate;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class CuponResponse {

    private Long id;
    private String codigo;
    private TipoDescuento tipoDescuento;
    private BigDecimal valor;
    private LocalDate fechaInicio;
    private LocalDate fechaFin;
    private Integer usoMaximo;
    private Integer usosActuales;
    private EstadoCupon estado;
}